edb/schema/functions.py

```diff
@@ -667,17 +666,27 @@ def is_blocking_ref(
         return not isinstance(reference, Parameter)
 
 
-class CallableCommand(sd.QualifiedObjectCommand):
+class CallableCommand(sd.QualifiedObjectCommand[CallableObject]):
 
-    def _get_params(self, schema: s_schema.Schema, context):
+    def _get_params(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext
+    ) -> so.ObjectList[Parameter]:
         params = []
         for cr_param in self.get_subcommands(type=ParameterCommand):
-            param = schema.get(cr_param.classname)
+            param: Parameter = schema.get(cr_param.classname)
             params.append(param)
         return FuncParameterList.create(schema, params)
 
-    def _alter_innards(self, schema: s_schema.Schema, context):
-        schema = super()._alter_innards(schema, context)
+    def _alter_innards(
```

Comment:
Can you define an `AlterCallableObject` and move this there instead?

Reply:
Don't forget to actually *use* the newly defined class by extending from it in the appropriate `AlterFunction` and `AlterOperator`